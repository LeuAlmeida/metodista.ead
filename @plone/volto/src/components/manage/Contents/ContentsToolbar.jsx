/**
 * Contents toolbar component.
 * @module components/manage/Contents/ContentsToolbar
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { Icon } from 'semantic-ui-react';
import { defineMessages, injectIntl, intlShape } from 'react-intl';

import { getBaseUrl } from '../../../helpers';

const messages = defineMessages({
  cancel: {
    id: 'Cancel',
    defaultMessage: 'Cancel',
  },
});

@injectIntl
@connect((state, props) => ({
  pathname: props.location.pathname,
}))
/**
 * ContentsToolbar container class.
 * @class ContentsToolbar
 * @extends Component
 */
export default class ContentsToolbar extends Component {
  /**
   * Property types.
   * @property {Object} propTypes Property types.
   * @static
   */
  static propTypes = {
    pathname: PropTypes.string.isRequired,
    intl: intlShape.isRequired,
  };

  /**
   * Render method.
   * @method render
   * @returns {string} Markup for the component.
   */
  render() {
    return (
      <div>
        <Link
          to={getBaseUrl(this.props.pathname)}
          id="toolbar-cancel"
          className="item"
        >
          <Icon
            name="remove"
            size="big"
            color="red"
            title={this.props.intl.formatMessage(messages.cancel)}
          />
        </Link>
      </div>
    );
  }
}
