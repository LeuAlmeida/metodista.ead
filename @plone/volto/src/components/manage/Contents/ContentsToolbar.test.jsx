import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';
import { MemoryRouter } from 'react-router-dom';

import ContentsToolbar from './ContentsToolbar';

const mockStore = configureStore();

describe('ContentsToolbar', () => {
  it('renders a contents toolbar component', () => {
    const store = mockStore({
      intl: {
        locale: 'en',
        messages: {},
      },
    });
    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <ContentsToolbar location={{ pathname: '/blog' }} />
        </MemoryRouter>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
