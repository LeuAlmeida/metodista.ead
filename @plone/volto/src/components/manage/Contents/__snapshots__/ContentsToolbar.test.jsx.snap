// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentsToolbar renders a contents toolbar component 1`] = `
<div>
  <a
    className="item"
    href="/blog"
    id="toolbar-cancel"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="red remove big icon"
      title="Cancel"
    />
  </a>
</div>
`;
